### PR TITLE
nrf_wifi: Add ops to set and get checksum status, manifest: nrfxlib

### PR DESCRIPTION
**Pull request by kapbh**
Add helpers to get/set checksum status in the network buffer. This are missing in linux driver.

### CODE REVIEW

**Review by ajayparida — APPROVED** [review]

**Comment by krish2718** [comment]
The following west manifest projects have been modified in this Pull Request:

| Name | Old Revision | New Revision | Diff |
| ---- | ------------ | ------------ |------|
| nrfxlib | https://github.com/nrfconnect/sdk-nrfxlib/commit/622afe102c933c660d20ef811822a71d1c016845 | https://github.com/nrfconnect/sdk-nrfxlib/pull/1258 | [nrfconnect/sdk-nrfxlib#1258/files](https://github.com/nrfconnect/sdk-nrfxlib/pull/1258/files) |

*Note: This message is automatically posted and updated by the Manifest GitHub Action.*

File: src/shim.c [hunk 1]
```diff
@@ -336,6 +336,20 @@ static unsigned char shim_nbuf_get_priority(void *nbuf)
 	return skb->priority;
 }
 
+static unsigned char shim_nbuf_get_chksum_done(void *nbuf)
+{
+	struct sk_buff *skb = (struct sk_buff *)nbuf;
+
```
Comment (sachinthegreen) [inline src/shim.c:342]: Might be prudent to put a `NULL` check for skb here.

File: src/shim.c [hunk 2]
```diff
@@ -336,6 +336,20 @@ static unsigned char shim_nbuf_get_priority(void *nbuf)
...
+	return skb->csum_complete_sw;
+}
+
+static void shim_nbuf_set_chksum_done(void *nbuf, unsigned char chksum_done)
+{
+	struct sk_buff *skb = (struct sk_buff *)nbuf;
+
```
Comment (sachinthegreen) [inline src/shim.c:349]: Might be prudent to put a `NULL` check for skb here.

**Review by sachinthegreen — CHANGES_REQUESTED** [review]

File: src/shim.c [hunk 3]
```diff
@@ -336,6 +336,28 @@ static unsigned char shim_nbuf_get_priority(void *nbuf)
...
+static unsigned char shim_nbuf_get_chksum_done(void *nbuf)
+{
+	struct sk_buff *skb = (struct sk_buff *)nbuf;
+
+	if (!skb) {
+		pr_err("%s: Unable to get checksum\n", __func__);
+		return -1;
+	}
+
+	return skb->csum_complete_sw;
```
Comment (krish2718) [inline src/shim.c:348]: In general this flag needs additional data, has this been tested?

```
  skb->csum = csum;
  skb->ip_summed = CHECKSUM_COMPLETE;
  skb->csum_complete_sw = 1;
  skb->csum_valid = !sum;
```

Comment (kapbh) [reply src/shim.c:348]: I have tested linux driver with RPI4 with this change. It's building and also connecting with AP.

Comment (krish2718) [reply src/shim.c:348]: With CSUM offload enabled and using IPv4 fragmentation?

Comment (kapbh) [reply src/shim.c:358]: Yes, need to add those flags.

**Review by ajayparida — APPROVED** [review]

File: Makefile [hunk 4]
```diff
@@ -114,18 +114,32 @@ ccflags-y += \
...
-	     -DCONFIG_NRF700X_BAND_UNII_1_UPPER_EDGE_BACKOFF=0 \
-	     -DCONFIG_NRF700X_BAND_UNII_2A_LOWER_EDGE_BACKOFF=0 \
-	     -DCONFIG_NRF700X_BAND_UNII_2A_UPPER_EDGE_BACKOFF=0 \
-	     -DCONFIG_NRF700X_BAND_UNII_2C_LOWER_EDGE_BACKOFF=0 \
-	     -DCONFIG_NRF700X_BAND_UNII_2C_UPPER_EDGE_BACKOFF=0 \
-	     -DCONFIG_NRF700X_BAND_UNII_3_LOWER_EDGE_BACKOFF=0 \
-	     -DCONFIG_NRF700X_BAND_UNII_3_UPPER_EDGE_BACKOFF=0 \
-	     -DCONFIG_NRF700X_BAND_UNII_4_LOWER_EDGE_BACKOFF=0 \
-	     -DCONFIG_NRF700X_BAND_UNII_4_UPPER_EDGE_BACKOFF=0 \
+	     -DCONFIG_NRF700X_BAND_2G_LOWER_EDGE_BACKOFF_DSSS=0 \
```
Comment (ajayparida) [inline Makefile:117]: Please add CONFIG_NRF700X_TCP_IP_CHECKSUM_OFFLOAD=y as by default for Zephyr this is enabled.

**Review by ajayparida — CHANGES_REQUESTED** [review]

**Review by ajayparida — APPROVED** [review]

**Review by sachinthegreen — APPROVED** [review]

**Pull request merged**